src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -620,10 +634,10 @@ export class HaConfigDevicePage extends LitElement {
             ${this._related?.script?.length
               ? html`
                   <div class="items">
-                    ${this._related.script.map((script) => {
-                      const entityState = this.hass.states[script];
+                    ${this._toEntities(this._related.script).map((script) => {
+                      const entityState = script;
                       const entry = this._entityReg.find(
-                        (e) => e.entity_id === script
+                        (e) => e.entity_id === (script as any)
```

Comment:
script is an `stateObj` right? So this should be?
```suggestion
                        (e) => e.entity_id === script.entity_id
```

Reply:
My bad! Fixed.